Java/core/src/main/java/com/amazon/randomcutforest/tree/BoxCache.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.tree;
+
+import java.util.HashMap;
+import java.util.Random;
+
+public class BoxCache<Point> implements IBoxCache<Point> {
+
+    double cacheFraction;
+    Random cacheRandom;
+    long randomSeed;
+    AbstractBoundingBox<Point>[] cachedBoxes;
+    HashMap<Integer, Integer> cacheMap;
+    int maxSize;
+
+    protected BoxCache(long seed, double cacheFraction, int maxSize) {
+        randomSeed = seed;
+        this.cacheFraction = cacheFraction;
+        this.maxSize = maxSize;
+    }
+
+    @Override
+    public boolean inUse(int index) {
+        if (cacheFraction == 0.0) {
+            return false;
+        } else if (cacheFraction == 1.0) {
+            return true;
+        } else if (cacheFraction >= 0.5) {
```

Comment:
I would extract this check into a private or protected method, `isDirectMap` or something like that.